main.go

```diff
@@ -184,6 +186,17 @@ var head = []string{
 	"<!doctype", // HTML doctype
 }
 
+func getExtension(s string) string {
```

Comment:
I'd call it fileExtension, avoiding usage of getXxx getters name as well as a doc comment to make things clear. For instance:

```go
// fileExtension returns file extension if non empty or base file name otherwise.
// The result is always lowercased
func fileExtension(name string) string
```

Reply:
I added lower case part, it wasn't there. Just thought it might be a good idea given how diverse naming of docker files is. Up to you though.

Reply:
Also, can we move it above `var head ...`?
Because `var head ...` is used in the function right below it, the new `getExtension` func is kind of getting in the way.

Reply:
Ack.